### PR TITLE
Improve Events and ServerMockup

### PR DESCRIPTION
**Pull request by RogerHardiman**
Events.....    Changes to make events work with Axis cameras and with NAT.

There is a new file called example6.js which I am using to connect to 4 different cameras and to the serverMockup and can receive events from all of them.

The Axis events are not documented properly in the ONVIF Spec and I had to use ODM and Wireshark. They need to include a <To> tag and a <SubscriptionId> tag in the Pull Request.

ServerMockup.... made changes so discovery works properly and ODM now detects it. Added a few other XML stubs. The purpose was to get ODM to receive events from the ServerMockup so I could verify how ODM received events.

### CODE REVIEW

File: example6.js [hunk 1]
```diff
@@ -36,14 +41,37 @@ new Cam({
 
     // Use Nimbe's flow to execute ONVIF commands in sequence
     flow.series([
```
Comment (agsh) [inline example6.js]: I think we can use promises here since engine version in package.json is ` "node": ">=6.0"`

File: package.json [hunk 2]
```diff
@@ -29,6 +29,11 @@
       "name": "Jeff Galbraith",
       "email": "jgalbraith@intelliviewtech.com",
       "url": "http://intelliviewtech.com"
+    },
+    {
+      "name": "Roger Hardiman",
+      "email": "opensource@rjh.org.uk",
+      "url": "http://www.rjh.org.uk"
```
Comment (agsh) [inline package.json]: 👍

**Comment by agsh** [comment]
Only two fixes from the CI left: https://travis-ci.org/agsh/onvif/builds/608839166#L229

File: example6.js (hunk 1 above)
Comment (RogerHardiman) [reply example6.js]: Good idea. All my other examples use nimble so I will have a go with promises on all of them. Could we merge without example6 and leave the example in my branch?

**Comment by RogerHardiman** [comment]
lint changes have now been committed to my branch

**Comment by RogerHardiman** [comment]
Hi @agsh 
The Travis CI builds are terminating again.
It may have been stuck as I added an extra UDP socket into serverMockup.coffee
To make the code cleaner I have implemented a close() in serverMockup and test called 'zzzterminate.coffee' so it is the last test executed (alphabetical order) and it closes the serverMockup.

**Comment by agsh** [comment]
> Hi @agsh
> The Travis CI builds are terminating again.
> It may have been stuck as I added an extra UDP socket into serverMockup.coffee
> To make the code cleaner I have implemented a close() in serverMockup and test called 'zzzterminate.coffee' so it is the last test executed (alphabetical order) and it closes the serverMockup.

Yes, I thought about this a year ago here: https://github.com/agsh/onvif/blob/v1/test/serverMockup.js when try to rewrite tests in js. But I never completed the job.

I think, you can merge this branch into the master. Thanks again!

**Pull request merged**